drivers/can/can_stm32.c

```diff
@@ -419,6 +426,10 @@ static int can_stm32_init(struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_CAN_2
```

Comment:
@galak, interesting example where DT_INST would face limitation.
CAN_1 and CAN_2 could not be enabled together and yet we need to have CAN_2 specific code.
Or maybe you have an alternate idea.

Reply:
Would need to understand a bit more about what FMR to see if we can figure out something workable.  But agreed, that DT_INST doesn't work here.  There are always a few cases in which having the SoC's concept of "number" is needed.

Reply:
Not sure what FMR stands for, but I get the general idea :-).